src/SAML2/XML/shibmd/Scope.php

```diff
@@ -24,7 +24,7 @@ class Scope
      *
      * @var string
      */
-    public $scope;
+    public $scope = '';
```

Comment:
Can the scope be an empty string?

Reply:
Consider this:

```php
            $scope = new \SAML2\XML\Shibmd\Scope();
            $string = $scope->toXML();
```

In that case, the `$scope` property will be set to an empty string, and that'll be reflected in the resulting XML. I don't know if the scope is mandatory or if it can be empty. However, there are two things we must consider:

* If the scope is not mandatory, we shouldn't be setting it to an empty string, but allow it to be `null` and skip it completely in `toXML()`. This means we should also allow clearing it, by allowing the argument to `setScope()` to be `null`.
* In that case, `getScope()` needs to be able to return `null` as well.

Reply:
Not sure about the standards.. An empty scope doesn't seem to make much sense, but I can't find a definitive answer anywhere.. The old `assert()` in toXML would only check if it's a string or not, so a null-value would've failed.

I still think it is a good idea to set the empty string and add an assertion in the toXML-method to fail on an empty string.

Reply:
Given that this is shibboleth-specific, there's probably no standard about it. In any case, I agree, an empty string as the default makes things much easier for us!